src/app/utils/BountyUtils.ts

```diff
@@ -262,6 +269,201 @@ const BountyUtils = {
+                components: [{
+                    type: 2,
+                    label: "Back to List",
+                    style: 5,
+                    url: customer.lastListMessage,
+                }]
+            }];
+        }
+
+        // Create/Update the card
+        let cardMessage: Message;
+
+        if (isDraftBounty) {  // If we are in Create (Draft) mode, put the card in the DM channel
+            cardMessage = await ( await DiscordUtils.getGuildMemberFromUserId(bounty.createdBy.discordId, bounty.customerId)).send(cardEmbeds);
+        } else {
+            if (activity == Activities.publish) {  // Publishing. If the card exists, delete it - it was in a DM}
+                if (!!bounty.canonicalCard) {
+                    const draftChannel = await DiscordUtils.getTextChannelfromChannelId(bounty.canonicalCard.channelId);
+                    const draftCardMessage = await DiscordUtils.getMessagefromMessageId(bounty.canonicalCard.messageId, draftChannel);
+                    await draftCardMessage.delete();
+                    bounty.canonicalCard = undefined;
+                }
+            }
+            if (!!bounty.canonicalCard) {  // If we still have an existing card, just edit it, remove old reactions
+                bountyChannel = await DiscordUtils.getTextChannelfromChannelId(bounty.canonicalCard.channelId);
+                cardMessage = await DiscordUtils.getMessagefromMessageId(bounty.canonicalCard.messageId, bountyChannel);
+                await cardMessage.edit(cardEmbeds);
+                await cardMessage.reactions.removeAll();
+            } else {  // Otherwise create it. Put it in the passed in channel, or customer channel by default
+                if (!bountyChannel) bountyChannel = await DiscordUtils.getBountyChannelfromCustomerId(bounty.customerId);
+                cardMessage = await bountyChannel.send(cardEmbeds);
+            }
+        }
+        reacts.forEach(react => {
+            cardMessage.react(react);
+        });
+
+
+        // Delete old cards if they exist. Notify user of new card location with link
+
+        if (bounty.discordMessageId) {
+            await this.notifyAndRemove(bounty.discordMessageId, await DiscordUtils.getBountyChannelfromCustomerId(bounty.customerId), cardMessage.url);
+        }
+        if (!!bounty.creatorMessage) {
+            await this.notifyAndRemove(bounty.creatorMessage.messageId, await DiscordUtils.getTextChannelfromChannelId(bounty.creatorMessage.channelId), cardMessage.url);
+        }
+        if (!!bounty.claimantMessage) {
+            await this.notifyAndRemove(bounty.claimantMessage.messageId, await DiscordUtils.getTextChannelfromChannelId(bounty.claimantMessage.channelId), cardMessage.url);
+        }
+         
+        // Store the card location in the bounty, remove the old cards
+        const writeResult: UpdateWriteOpResult = await bountyCollection.updateOne({ _id: new mongo.ObjectId(bounty._id) }, {
```

Comment:
I think I understand why you put updating the message store logic in this method. I'm not a huge fan of how the method name obscures the action of updating the message store (critical for activities and client sync) and putting two logically different operations in one util method. 

If you agree, I think a good strategy would be to break the db call into it's own method in BountyUtils, and then create a third method which calls both `BountyUtils.canonicalCard` and `BountyUtils.updateMessageStore`

Reply:
I agree about taking #3 out since it writes to the DB. I think the other two belong together, however, since they essentially update the UI in one step.

Reply:
Done